### PR TITLE
GEODE-5405 fix backup tests

### PR DESCRIPTION
**Pull request by kirklund**
4 commits:

commit 218ddc92417ba870e072d673cf6d02748ca18152
Author: Kirk Lund <klund@apache.org>
Date:   Thu Jul 12 16:59:55 2018 -0700

    GEODE-5405: Streamline Backup distributed tests

commit 772bac0b157fec8090975830acd23cd548b8b556
Author: Kirk Lund <klund@apache.org>
Date:   Thu Jul 12 16:58:28 2018 -0700

    GEODE-5405: Use zero as default revokedTime in PersistentMemberPattern
    
    Different code paths were using different default. One was zero while
    another was -1. This prevent comparisons including collection contains
    operations.

commit d6b32d8166f8d452068809b15b6aa6dcf903a6b9
Author: Kirk Lund <klund@apache.org>
Date:   Thu Jul 12 16:56:21 2018 -0700

    GEODE-5431: Improve error message when using DistributedDiskDirRule with VM.bounce()

commit c2b9122a7be2a96e086a65830bed2ab70313a3b6
Author: Kirk Lund <klund@apache.org>
Date:   Thu Jul 12 16:54:17 2018 -0700

    GEODE-5430: Add getController method to VM

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistentMemberPattern.java [hunk 1]
```diff
@@ -39,19 +39,19 @@
   protected long revokedTime;
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistentMemberPattern.java]: I noticed all the instance variables in the class are protected even though they can be private

Comment (kirklund) [reply geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistentMemberPattern.java]: I pushed another commit that changes the fields to private and adds in missing `@Override` annotations.

**Comment by kirklund** [comment]
I'm getting this failure intermittently:

org.apache.geode.test.dunit.RMIException: While invoking org.apache.geode.internal.cache.backup.BackupDistributedTest$$Lambda$84/1532139270.run in VM 0 running on Host 10.118.33.216 with 4 VMs

	at org.apache.geode.test.dunit.VM.invoke(VM.java:443)
	at org.apache.geode.test.dunit.VM.invoke(VM.java:412)
	at org.apache.geode.test.dunit.VM.invoke(VM.java:355)
	at org.apache.geode.internal.cache.backup.BackupDistributedTest.testRecoverySystemWithConcurrentPutter(BackupDistributedTest.java:452)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.apache.geode.test.dunit.rules.AbstractDistributedTestRule$1.evaluate(AbstractDistributedTestRule.java:60)
	at org.apache.geode.test.junit.rules.DescribedExternalResource$1.evaluate(DescribedExternalResource.java:40)
	at org.apache.geode.test.dunit.rules.AbstractDistributedTestRule$1.evaluate(AbstractDistributedTestRule.java:60)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at junitparams.JUnitParamsRunner.runChild(JUnitParamsRunner.java:449)
	at junitparams.JUnitParamsRunner.runChild(JUnitParamsRunner.java:393)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:67)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: org.apache.geode.cache.PartitionedRegionStorageException: Unable to find any members to host a bucket in the partitioned region. Advise you to start enough data store nodes.
Partitioned Region name = /BackupDistributedTest_testRecoverySystemWithConcurrentPutter_region1
   Redundancy level set to 0
. Number of available data stores: 0
   . Number successfully allocated = 0
. Data stores: []
. Data stores successfully allocated: []
. Equivalent members: [10.118.33.216(91287)<v33>:32772, 10.118.33.216(91286)<v33>:32770, 10.118.33.216(91285)<v33>:32771, 10.118.33.216(91284:locator)<ec><v0>:32769]

	at org.apache.geode.internal.cache.PRHARedundancyProvider.insufficientStores(PRHARedundancyProvider.java:306)
	at org.apache.geode.internal.cache.PRHARedundancyProvider.createBucketOnDataStore(PRHARedundancyProvider.java:478)
	at org.apache.geode.internal.cache.PartitionedRegion.createBucket(PartitionedRegion.java:3334)
	at org.apache.geode.internal.cache.PartitionedRegion.virtualPut(PartitionedRegion.java:2075)
	at org.apache.geode.internal.cache.LocalRegionDataView.putEntry(LocalRegionDataView.java:152)
	at org.apache.geode.internal.cache.LocalRegion.basicPut(LocalRegion.java:5086)
	at org.apache.geode.internal.cache.LocalRegion.validatedPut(LocalRegion.java:1618)
	at org.apache.geode.internal.cache.LocalRegion.put(LocalRegion.java:1605)
	at org.apache.geode.internal.cache.AbstractRegion.put(AbstractRegion.java:413)
	at org.apache.geode.internal.cache.backup.BackupDistributedTest.createData(BackupDistributedTest.java:699)
	at org.apache.geode.internal.cache.backup.BackupDistributedTest.lambda$testRecoverySystemWithConcurrentPutter$bb17a952$4(BackupDistributedTest.java:453)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hydra.MethExecutor.executeObject(MethExecutor.java:244)
	at org.apache.geode.test.dunit.standalone.RemoteDUnitVM.executeMethodOnObject(RemoteDUnitVM.java:70)
	at sun.reflect.GeneratedMethodAccessor15.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:357)
	at sun.rmi.transport.Transport$1.run(Transport.java:200)
	at sun.rmi.transport.Transport$1.run(Transport.java:197)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.Transport.serviceCall(Transport.java:196)
	at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:573)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:835)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:688)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:687)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**Comment by kirklund** [comment]
I found the cause. The 1st lines are the datastores creating cache + regions async. This should be synchronous:
```java
    vm1.invoke(() -> createColocatedRegions(vm1));
    vm2.invoke(() -> createColocatedRegions(vm2));

    vm0.invoke(() -> createAccessor());
```

**Pull request merged**